crates/core/src/operations/vacuum.rs

```diff
@@ -235,6 +245,31 @@ impl VacuumBuilder {
             None => Utc::now().timestamp_millis(),
         };
 
+        let keep_files = match &self.keep_versions {
+            Some(versions) => {
+                let mut keep_files: HashSet<String> = HashSet::new();
+                let mut state =
+                    DeltaTableState::try_new(&self.log_store, DeltaTableConfig::default(), Some(0))
+                        .await?;
+                let mut sorted_versions = versions.clone();
+                sorted_versions.sort();
+                for version in sorted_versions {
+                    if version == 0 {
+                        continue;
+                    }
```

Comment:
I think this continue isn't needed, we might need to keep some files from version 0, and it looks like `state.update` handles updating to the version it's already on and will return early.

Reply:
Actually I think it is needed. I am basing this logic on how load_version is written which uses update_incremental. See: https://github.com/delta-io/delta-rs/blob/eff0b582a7e31242494b9ba77a0441174c6c2227/crates/core/src/table/mod.rs#L210
I want to follow the logic they use there. From my reading, it seems version 0 cannot have files.